### PR TITLE
Expose new 'root' attribute, pointing to ultimate ancestor

### PR DESCRIPTION
**Pull request by danielballan**
This, plus a forthcoming bluesky PR, solves the problem that #280 meant to solve, better. Thanks, @klauer.

### CODE REVIEW

File: ophyd/device.py [hunk 1]
```diff
@@ -348,6 +348,15 @@ def read(self):
     def describe(self):
         return OrderedDict()
 
+    @property
+    def root(self):
```
Comment (klauer) [inline ophyd/device.py]: `Device` is an `OphydObject` - do we need this on both?

File: ophyd/device.py [hunk 2]
```diff
@@ -805,3 +816,7 @@ def _repr_info(self):
         yield ('read_attrs', self.read_attrs)
         yield ('configuration_attrs', self.configuration_attrs)
         yield ('monitor_attrs', self.monitor_attrs)
+
+
+class RedundantStaging(Exception):
```
Comment (klauer) [inline ophyd/device.py]: Can you please move this to `ophyd.utils.errors` and subclass from `OpException`?

**Comment by danielballan** [comment]
See https://github.com/NSLS-II/bluesky/pull/369

File: ophyd/device.py (hunk 2 above)
Comment (danielballan) [inline ophyd/device.py]: Good call. Done.

**Comment by klauer** [comment]
:+1:

**Pull request merged**